<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Visualization of Functional Enrichment Result | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="Chapter 12 Visualization of Functional Enrichment Result | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Visualization of Functional Enrichment Result | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Visualization of Functional Enrichment Result | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter11.html"/>
<link rel="next" href="chapter13.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.9/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#terminology"><i class="fa fa-check"></i><b>1.1</b> Terminology</a><ul>
<li class="chapter" data-level="1.1.1" data-path="chapter1.html"><a href="chapter1.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>1.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="1.1.2" data-path="chapter1.html"><a href="chapter1.html#gene-ontology-go"><i class="fa fa-check"></i><b>1.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="1.1.3" data-path="chapter1.html"><a href="chapter1.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>1.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="1.1.4" data-path="chapter1.html"><a href="chapter1.html#other-gene-sets"><i class="fa fa-check"></i><b>1.1.4</b> Other gene sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Functional Enrichment Analysis Methods</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#over-representation-analysis"><i class="fa fa-check"></i><b>2.1</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.3</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#input-data"><i class="fa fa-check"></i><b>3.1</b> Input data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#cell-marker"><i class="fa fa-check"></i><b>3.3</b> Cell Marker</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#msigdb-analysis"><i class="fa fa-check"></i><b>3.4</b> MSigDb analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="chapter4.html"><a href="chapter4.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Gene Ontology Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter5.html"><a href="chapter5.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="chapter5.html"><a href="chapter5.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chapter5.html"><a href="chapter5.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>5.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="chapter12.html"><a href="chapter12.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="chapter12.html"><a href="chapter12.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="chapter12.html"><a href="chapter12.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="chapter12.html"><a href="chapter12.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="chapter12.html"><a href="chapter12.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> dplyr verbs for clusterProfiler</a><ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#filter"><i class="fa fa-check"></i><b>13.1</b> filter</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#arrange"><i class="fa fa-check"></i><b>13.2</b> arrange</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#select"><i class="fa fa-check"></i><b>13.3</b> select</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#mutate"><i class="fa fa-check"></i><b>13.4</b> mutate</a></li>
<li class="chapter" data-level="13.5" data-path="chapter13.html"><a href="chapter13.html#slice"><i class="fa fa-check"></i><b>13.5</b> slice</a></li>
<li class="chapter" data-level="13.6" data-path="chapter13.html"><a href="chapter13.html#summarise"><i class="fa fa-check"></i><b>13.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="chapter14.html"><a href="chapter14.html"><i class="fa fa-check"></i><b>14</b> Useful utilities</a><ul>
<li class="chapter" data-level="14.1" data-path="chapter14.html"><a href="chapter14.html#bitr"><i class="fa fa-check"></i><b>14.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="14.1.1" data-path="chapter14.html"><a href="chapter14.html#bitr_kegg"><i class="fa fa-check"></i><b>14.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="chapter14.html"><a href="chapter14.html#setReadable"><i class="fa fa-check"></i><b>14.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter12" class="section level1">
<h1><span class="header-section-number">Chapter 12</span> Visualization of Functional Enrichment Result</h1>
<p>The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package implements several visualization methods
to help interpreting enrichment results. It supports visualizing enrichment
results obtained from <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(Yu et al. <a href="#ref-yu_dose_2015" role="doc-biblioref">2015</a>)</span>,
<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> <span class="citation">(Yu et al. <a href="#ref-yu2012" role="doc-biblioref">2012</a>)</span>,
<a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> <span class="citation">(Yu and He <a href="#ref-yu_reactomepa_2016" role="doc-biblioref">2016</a>)</span> and <a href="http://bioconductor.org/packages/meshes">meshes</a>. Both
over representation analysis (ORA) and gene set enrichment analysis (GSEA) are
supported.</p>
<p>Many of these visualization methods were first implemented in <a href="http://bioconductor.org/packages/DOSE">DOSE</a> and rewrote from scratch using <code>ggplot2</code>. If you want to use old methods<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, you can use the <a href="https://github.com/GuangchuangYu/doseplot">doseplot</a> package.</p>
<div id="bar-plot" class="section level2">
<h2><span class="header-section-number">12.1</span> Bar Plot</h2>
<p>Bar plot is the most widely used method to visualize enriched terms. It depicts
the enrichment scores (<em>e.g.</em> p values) and gene count or ratio as bar height
and color.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="chapter12.html#cb93-1"></a><span class="kw">library</span>(DOSE)</span>
<span id="cb93-2"><a href="chapter12.html#cb93-2"></a><span class="kw">data</span>(geneList)</span>
<span id="cb93-3"><a href="chapter12.html#cb93-3"></a>de &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb93-4"><a href="chapter12.html#cb93-4"></a></span>
<span id="cb93-5"><a href="chapter12.html#cb93-5"></a>edo &lt;-<span class="st"> </span><span class="kw">enrichDGN</span>(de)</span></code></pre></div>


<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="chapter12.html#cb94-1"></a><span class="kw">library</span>(enrichplot)</span>
<span id="cb94-2"><a href="chapter12.html#cb94-2"></a><span class="kw">barplot</span>(edo, <span class="dt">showCategory=</span><span class="dv">20</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Barplot"></span>
<img src="clusterProfiler_files/figure-html/Barplot-1.png" alt="Bar plot of enriched terms." width="768" />
<p class="caption">
Figure 12.1: <strong>Bar plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="dot-plot" class="section level2">
<h2><span class="header-section-number">12.2</span> Dot plot</h2>
<p>Dot plot is similar to bar plot with the capability to encode another score as
dot size.</p>


<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="chapter12.html#cb95-1"></a>edo2 &lt;-<span class="st"> </span><span class="kw">gseNCG</span>(geneList, <span class="dt">nPerm=</span><span class="dv">10000</span>)</span>
<span id="cb95-2"><a href="chapter12.html#cb95-2"></a>p1 &lt;-<span class="st"> </span><span class="kw">dotplot</span>(edo, <span class="dt">showCategory=</span><span class="dv">30</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;dotplot for ORA&quot;</span>)</span>
<span id="cb95-3"><a href="chapter12.html#cb95-3"></a>p2 &lt;-<span class="st"> </span><span class="kw">dotplot</span>(edo2, <span class="dt">showCategory=</span><span class="dv">30</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;dotplot for GSEA&quot;</span>)</span>
<span id="cb95-4"><a href="chapter12.html#cb95-4"></a><span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:Dotplotcap"></span>
<img src="clusterProfiler_files/figure-html/Dotplotcap-1.png" alt="Dot plot of enriched terms." width="1152" />
<p class="caption">
Figure 12.2: <strong>Dot plot of enriched terms.</strong>
</p>
</div>
</div>
<div id="gene-concept-network" class="section level2">
<h2><span class="header-section-number">12.3</span> Gene-Concept Network</h2>
<p>Both the <code>barplot</code> and <code>dotplot</code> only displayed most significant enriched terms,
while users may want to know which genes are involved in these significant
terms.
In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories and provide information of numeric changes if available, we developed <code>cnetplot</code> function to extract the complex association.
The <code>cnetplot</code> depicts the linkages of genes and biological concepts (<em>e.g.</em> GO terms or KEGG pathways) as a network. GSEA result is also supported
with only core enriched genes displayed.</p>


<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="chapter12.html#cb96-1"></a><span class="co">## convert gene ID to Symbol</span></span>
<span id="cb96-2"><a href="chapter12.html#cb96-2"></a>edox &lt;-<span class="st"> </span><span class="kw">setReadable</span>(edo, <span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="st">&#39;ENTREZID&#39;</span>)</span>
<span id="cb96-3"><a href="chapter12.html#cb96-3"></a>p1 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">foldChange=</span>geneList)</span>
<span id="cb96-4"><a href="chapter12.html#cb96-4"></a><span class="co">## categorySize can be scaled by &#39;pvalue&#39; or &#39;geneNum&#39;</span></span>
<span id="cb96-5"><a href="chapter12.html#cb96-5"></a>p2 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">categorySize=</span><span class="st">&quot;pvalue&quot;</span>, <span class="dt">foldChange=</span>geneList)</span>
<span id="cb96-6"><a href="chapter12.html#cb96-6"></a>p3 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">foldChange=</span>geneList, <span class="dt">circular =</span> <span class="ot">TRUE</span>, <span class="dt">colorEdge =</span> <span class="ot">TRUE</span>)</span>
<span id="cb96-7"><a href="chapter12.html#cb96-7"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, <span class="dt">ncol=</span><span class="dv">3</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="dt">rel_widths=</span><span class="kw">c</span>(.<span class="dv">8</span>, <span class="fl">.8</span>, <span class="fl">1.2</span>))</span></code></pre></div>
<div class="figure"><span id="fig:Networkplot"></span>
<img src="clusterProfiler_files/figure-html/Networkplot-1.png" alt="Network plot of enriched terms." width="1920" />
<p class="caption">
Figure 12.3: <strong>Network plot of enriched terms.</strong>
</p>
</div>
<p>If you would like label subset of the nodes, you can use the <code>node_label</code> parameter, which supports 4 possible selections (i.e. “category”, “gene”, “all” and “none”), as demonstrated in Figure <a href="chapter12.html#fig:cnetNodeLabel">12.4</a>.</p>


<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="chapter12.html#cb97-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">node_label=</span><span class="st">&quot;category&quot;</span>) </span>
<span id="cb97-2"><a href="chapter12.html#cb97-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">node_label=</span><span class="st">&quot;gene&quot;</span>) </span>
<span id="cb97-3"><a href="chapter12.html#cb97-3"></a>p3 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">node_label=</span><span class="st">&quot;all&quot;</span>) </span>
<span id="cb97-4"><a href="chapter12.html#cb97-4"></a>p4 &lt;-<span class="st"> </span><span class="kw">cnetplot</span>(edox, <span class="dt">node_label=</span><span class="st">&quot;none&quot;</span>) </span>
<span id="cb97-5"><a href="chapter12.html#cb97-5"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, p4, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:cnetNodeLabel"></span>
<img src="clusterProfiler_files/figure-html/cnetNodeLabel-1.png" alt="Labelling nodes by selected subset. gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D)." width="1536" />
<p class="caption">
Figure 12.4: <strong>Labelling nodes by selected subset.</strong> gene category (A), gene name (B), both gene category and gene name (C, default) and not to label at all (D).
</p>
</div>
</div>
<div id="heatmap-like-functional-classification" class="section level2">
<h2><span class="header-section-number">12.4</span> Heatmap-like functional classification</h2>
<p>The <code>heatplot</code> is similar to <code>cnetplot</code>, while displaying the relationships as a
heatmap. The gene-concept network may become too complicated if user want to
show a large number significant terms. The <code>heatplot</code> can simplify the result
and more easy to identify expression patterns.</p>


<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="chapter12.html#cb98-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">heatplot</span>(edox)</span>
<span id="cb98-2"><a href="chapter12.html#cb98-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">heatplot</span>(edox, <span class="dt">foldChange=</span>geneList)</span>
<span id="cb98-3"><a href="chapter12.html#cb98-3"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">1</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Heatplot"></span>
<img src="clusterProfiler_files/figure-html/Heatplot-1.png" alt="Heatmap plot of enriched terms. default (A), foldChange=geneList (B)" width="1536" />
<p class="caption">
Figure 12.5: <strong>Heatmap plot of enriched terms.</strong> default (A), <code>foldChange=geneList</code> (B)
</p>
</div>
</div>
<div id="enrichment-map" class="section level2">
<h2><span class="header-section-number">12.5</span> Enrichment Map</h2>
<p>Enrichment map organizes enriched terms into a network with edges connecting
overlapping gene sets. In this way, mutually overlapping gene sets are tend to
cluster together, making it easy to identify functional module.</p>
<p>The <code>emapplot</code> function supports results obtained from hypergeometric test and gene set enrichment analysis. The <code>pie_scale</code> parameter can be used to resize nodes, as demonstrated in Figure <a href="chapter12.html#fig:Enrichment">12.6</a> B, and the <code>layout</code> parameter can adjust the layout, as demonstrated in Figure <a href="chapter12.html#fig:Enrichment">12.6</a> C and D.</p>


<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="chapter12.html#cb99-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(edo)</span>
<span id="cb99-2"><a href="chapter12.html#cb99-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(edo, <span class="dt">pie_scale=</span><span class="fl">1.5</span>)</span>
<span id="cb99-3"><a href="chapter12.html#cb99-3"></a>p3 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(edo,<span class="dt">layout=</span><span class="st">&quot;kk&quot;</span>)</span>
<span id="cb99-4"><a href="chapter12.html#cb99-4"></a>p4 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(edo, <span class="dt">pie_scale=</span><span class="fl">1.5</span>,<span class="dt">layout=</span><span class="st">&quot;kk&quot;</span>) </span>
<span id="cb99-5"><a href="chapter12.html#cb99-5"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, p4, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Enrichment"></span>
<img src="clusterProfiler_files/figure-html/Enrichment-1.png" alt="Plot for results obtained from hypergeometric test and gene set enrichment analysis. default (A), pie_scale=1.5 (B), layout="kk" (C) and pie_scale=1.5,layout="kk" (D)." width="1536" />
<p class="caption">
Figure 12.6: <strong>Plot for results obtained from hypergeometric test and gene set enrichment analysis.</strong> default (A), <code>pie_scale=1.5</code> (B), <code>layout="kk"</code> (C) and <code>pie_scale=1.5,layout="kk"</code> (D).
</p>
</div>
<p>The <code>emapplot</code> function also supports results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package. In addition to <code>pie_scale</code> and <code>layout</code> parameters, the number of circles in the bottom left corner can be adjusted using the <code>legend_n</code> parameteras, as demonstrated in Figure <a href="chapter12.html#fig:Enrichment2">12.7</a> B. And proportion of clusters in the pie chart can be adjusted using the <code>pie</code> parameter, when <code>pie="count"</code>, the proportion of clusters in the pie chart is determined by the number of genes, as demonstrated in Figure <a href="chapter12.html#fig:Enrichment2">12.7</a> C and D.</p>


<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="chapter12.html#cb100-1"></a><span class="kw">library</span>(clusterProfiler)</span>
<span id="cb100-2"><a href="chapter12.html#cb100-2"></a><span class="kw">data</span>(gcSample)</span>
<span id="cb100-3"><a href="chapter12.html#cb100-3"></a>xx &lt;-<span class="st"> </span><span class="kw">compareCluster</span>(gcSample, <span class="dt">fun=</span><span class="st">&quot;enrichKEGG&quot;</span>,</span>
<span id="cb100-4"><a href="chapter12.html#cb100-4"></a>                     <span class="dt">organism=</span><span class="st">&quot;hsa&quot;</span>, <span class="dt">pvalueCutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb100-5"><a href="chapter12.html#cb100-5"></a>p1 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(xx)</span>
<span id="cb100-6"><a href="chapter12.html#cb100-6"></a>p2 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(xx,<span class="dt">legend_n=</span><span class="dv">2</span>) </span>
<span id="cb100-7"><a href="chapter12.html#cb100-7"></a>p3 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(xx,<span class="dt">pie=</span><span class="st">&quot;count&quot;</span>)</span>
<span id="cb100-8"><a href="chapter12.html#cb100-8"></a>p4 &lt;-<span class="st"> </span><span class="kw">emapplot</span>(xx,<span class="dt">pie=</span><span class="st">&quot;count&quot;</span>, <span class="dt">pie_scale=</span><span class="fl">1.5</span>, <span class="dt">layout=</span><span class="st">&quot;kk&quot;</span>)</span>
<span id="cb100-9"><a href="chapter12.html#cb100-9"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, p4, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</span></code></pre></div>
<div class="figure"><span id="fig:Enrichment2"></span>
<img src="clusterProfiler_files/figure-html/Enrichment2-1.png" alt="Plot for results obtained from compareCluster function of clusterProfiler package. default (A), legend_n=2 (B), pie="count" (C) and pie="count", pie_scale=1.5, layout="kk" (D)." width="1536" />
<p class="caption">
Figure 12.7: <strong>Plot for results obtained from <code>compareCluster</code> function of <code>clusterProfiler</code> package.</strong> default (A), <code>legend_n=2</code> (B), <code>pie="count"</code> (C) and <code>pie="count", pie_scale=1.5, layout="kk"</code> (D).
</p>
</div>
</div>
<div id="upset-plot" class="section level2">
<h2><span class="header-section-number">12.6</span> UpSet Plot</h2>
<p>The <code>upsetplot</code> is an alternative to <code>cnetplot</code> for visualizing the complex
association between genes and gene sets. It emphasizes the gene overlapping
among different gene sets.</p>


<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="chapter12.html#cb101-1"></a><span class="kw">upsetplot</span>(edo)</span></code></pre></div>
<div class="figure"><span id="fig:upsetORA"></span>
<img src="clusterProfiler_files/figure-html/upsetORA-1.png" alt="Upsetplot for over-representation analysis." width="1152" />
<p class="caption">
Figure 12.8: <strong>Upsetplot for over-representation analysis.</strong>
</p>
</div>
<p>For over-representation analysis, <code>upsetplot</code> will calculate the overlaps among different gene sets as demonstrated in Figure <a href="chapter12.html#fig:upsetORA">12.8</a>. For GSEA result, it will plot the fold change distributions of different categories (e.g. unique to pathway, overlaps among different pathways).</p>


<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="chapter12.html#cb102-1"></a><span class="kw">upsetplot</span>(kk2) </span></code></pre></div>
<div class="figure"><span id="fig:upsetGSEA"></span>
<img src="clusterProfiler_files/figure-html/upsetGSEA-1.png" alt="Upsetplot for gene set enrichment analysis." width="1152" />
<p class="caption">
Figure 12.9: <strong>Upsetplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="ridgeline-plot-for-expression-distribution-of-gsea-result" class="section level2">
<h2><span class="header-section-number">12.7</span> ridgeline plot for expression distribution of GSEA result</h2>
<p>The <code>ridgeplot</code> will visualize expression distributions of core enriched genes
for GSEA enriched categories. It helps users to interpret up/down-regulated pathways.</p>


<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="chapter12.html#cb103-1"></a><span class="kw">ridgeplot</span>(edo2)</span></code></pre></div>
<div class="figure"><span id="fig:ridgeplot"></span>
<img src="clusterProfiler_files/figure-html/ridgeplot-1.png" alt="Ridgeplot for gene set enrichment analysis." width="1152" />
<p class="caption">
Figure 12.10: <strong>Ridgeplot for gene set enrichment analysis.</strong>
</p>
</div>
</div>
<div id="running-score-and-preranked-list-of-gsea-result" class="section level2">
<h2><span class="header-section-number">12.8</span> running score and preranked list of GSEA result</h2>
<p>Running score and preranked list are traditional methods for visualizing GSEA
result. The <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package supports both of them to visualize
the distribution of the gene set and the enrichment score.</p>


<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="chapter12.html#cb104-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">gseaplot</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span>, <span class="dt">by =</span> <span class="st">&quot;runningScore&quot;</span>, <span class="dt">title =</span> edo2<span class="op">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb104-2"><a href="chapter12.html#cb104-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">gseaplot</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span>, <span class="dt">by =</span> <span class="st">&quot;preranked&quot;</span>, <span class="dt">title =</span> edo2<span class="op">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb104-3"><a href="chapter12.html#cb104-3"></a>p3 &lt;-<span class="st"> </span><span class="kw">gseaplot</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span>, <span class="dt">title =</span> edo2<span class="op">$</span>Description[<span class="dv">1</span>])</span>
<span id="cb104-4"><a href="chapter12.html#cb104-4"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, <span class="dt">ncol=</span><span class="dv">1</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot"></span>
<img src="clusterProfiler_files/figure-html/gseaplot-1.png" alt="gseaplot for GSEA result(by = "runningScore"). by = "runningScore" (A), by = "preranked" (B), default (C)" width="1152" />
<p class="caption">
Figure 12.11: <strong>gseaplot for GSEA result(<code>by = "runningScore"</code>).</strong> <code>by = "runningScore"</code> (A), <code>by = "preranked"</code> (B), default (C)
</p>
</div>
<p>Another method to plot GSEA result is the <code>gseaplot2</code> function:</p>


<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="chapter12.html#cb105-1"></a><span class="kw">gseaplot2</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span>, <span class="dt">title =</span> edo2<span class="op">$</span>Description[<span class="dv">1</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot2"></span>
<img src="clusterProfiler_files/figure-html/gseaplot2-1.png" alt="Gseaplot2 for GSEA result." width="1152" />
<p class="caption">
Figure 12.12: <strong>Gseaplot2 for GSEA result.</strong>
</p>
</div>
<p>The <code>gseaplot2</code> also supports multile gene sets to be displayed on the same figure:</p>


<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="chapter12.html#cb106-1"></a><span class="kw">gseaplot2</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot22"></span>
<img src="clusterProfiler_files/figure-html/gseaplot22-1.png" alt="Gseaplot2 for GSEA result of multile gene sets." width="1152" />
<p class="caption">
Figure 12.13: <strong>Gseaplot2 for GSEA result of multile gene sets.</strong>
</p>
</div>
<p>User can also displaying the pvalue table on the plot via <code>pvalue_table</code>
parameter:</p>


<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="chapter12.html#cb107-1"></a><span class="kw">gseaplot2</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">pvalue_table =</span> <span class="ot">TRUE</span>,</span>
<span id="cb107-2"><a href="chapter12.html#cb107-2"></a>          <span class="dt">color =</span> <span class="kw">c</span>(<span class="st">&quot;#E495A5&quot;</span>, <span class="st">&quot;#86B875&quot;</span>, <span class="st">&quot;#7DB0DD&quot;</span>), <span class="dt">ES_geom =</span> <span class="st">&quot;dot&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot23"></span>
<img src="clusterProfiler_files/figure-html/gseaplot23-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add pvalue_table)." width="1152" />
<p class="caption">
Figure 12.14: <strong>Gseaplot2 for GSEA result of multile gene sets(add pvalue_table).</strong>
</p>
</div>
<p>User can specify <code>subplots</code> to only display a subset of plots:</p>


<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="chapter12.html#cb108-1"></a>p1 &lt;-<span class="st"> </span><span class="kw">gseaplot2</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">subplots =</span> <span class="dv">1</span>)</span>
<span id="cb108-2"><a href="chapter12.html#cb108-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">gseaplot2</span>(edo2, <span class="dt">geneSetID =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">subplots =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</span>
<span id="cb108-3"><a href="chapter12.html#cb108-3"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">1</span>, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gseaplot24"></span>
<img src="clusterProfiler_files/figure-html/gseaplot24-1.png" alt="Gseaplot2 for GSEA result of multile gene sets(add subplots). subplots = 1 (A),subplots = 1:2 (B)" width="1152" />
<p class="caption">
Figure 12.15: <strong>Gseaplot2 for GSEA result of multile gene sets(add subplots).</strong> <code>subplots = 1</code> (A),<code>subplots = 1:2</code> (B)
</p>
</div>
<p>The <code>gsearank</code> function plot the ranked list of genes belong to the specific
gene set.</p>


<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="chapter12.html#cb109-1"></a><span class="kw">gsearank</span>(edo2, <span class="dv">1</span>, <span class="dt">title =</span> edo2[<span class="dv">1</span>, <span class="st">&quot;Description&quot;</span>])</span></code></pre></div>
<div class="figure"><span id="fig:gsearank"></span>
<img src="clusterProfiler_files/figure-html/gsearank-1.png" alt="Ranked list of genes belong to the specific gene set." width="768" />
<p class="caption">
Figure 12.16: <strong>Ranked list of genes belong to the specific gene set.</strong>
</p>
</div>
<p>Multiple gene sets can be aligned using <code>cowplot</code>:
(ref:gsearank2scap) Gsearank for multiple gene sets.</p>

<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="chapter12.html#cb110-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb110-2"><a href="chapter12.html#cb110-2"></a><span class="kw">library</span>(cowplot)</span>
<span id="cb110-3"><a href="chapter12.html#cb110-3"></a></span>
<span id="cb110-4"><a href="chapter12.html#cb110-4"></a>pp &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="cf">function</span>(i) {</span>
<span id="cb110-5"><a href="chapter12.html#cb110-5"></a>    anno &lt;-<span class="st"> </span>edo2[i, <span class="kw">c</span>(<span class="st">&quot;NES&quot;</span>, <span class="st">&quot;pvalue&quot;</span>, <span class="st">&quot;p.adjust&quot;</span>)]</span>
<span id="cb110-6"><a href="chapter12.html#cb110-6"></a>    lab &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">names</span>(anno), <span class="st">&quot;=&quot;</span>,  <span class="kw">round</span>(anno, <span class="dv">3</span>), <span class="dt">collapse=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb110-7"><a href="chapter12.html#cb110-7"></a></span>
<span id="cb110-8"><a href="chapter12.html#cb110-8"></a>    <span class="kw">gsearank</span>(edo2, i, edo2[i, <span class="dv">2</span>]) <span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="ot">NULL</span>) <span class="op">+</span><span class="kw">ylab</span>(<span class="ot">NULL</span>) <span class="op">+</span></span>
<span id="cb110-9"><a href="chapter12.html#cb110-9"></a><span class="st">        </span><span class="kw">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="dv">0</span>, edo2[i, <span class="st">&quot;enrichmentScore&quot;</span>] <span class="op">*</span><span class="st"> </span><span class="fl">.9</span>, <span class="dt">label =</span> lab, <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">vjust=</span><span class="dv">0</span>)</span>
<span id="cb110-10"><a href="chapter12.html#cb110-10"></a>})</span>
<span id="cb110-11"><a href="chapter12.html#cb110-11"></a><span class="kw">plot_grid</span>(<span class="dt">plotlist=</span>pp, <span class="dt">ncol=</span><span class="dv">1</span>)</span></code></pre></div>
<div class="figure"><span id="fig:gsearank2"></span>
<img src="clusterProfiler_files/figure-html/gsearank2-1.png" alt="Gsearank for multiple gene sets." width="768" />
<p class="caption">
Figure 12.17: <strong>Gsearank for multiple gene sets.</strong>
</p>
</div>
</div>
<div id="pubmed-trend-of-enriched-terms" class="section level2">
<h2><span class="header-section-number">12.9</span> pubmed trend of enriched terms</h2>
<p>One of the problem of enrichment analysis is to find pathways for further
investigation. Here, we provide <code>pmcplot</code> function to plot the number/proportion
of publications trend based on the query result from PubMed Central. Of course,
users can use <code>pmcplot</code> in other scenarios. All text that can be queried on PMC
is valid as input of <code>pmcplot</code>.</p>


<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="chapter12.html#cb111-1"></a>terms &lt;-<span class="st"> </span>edo<span class="op">$</span>Description[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span>
<span id="cb111-2"><a href="chapter12.html#cb111-2"></a>p &lt;-<span class="st"> </span><span class="kw">pmcplot</span>(terms, <span class="dv">2010</span><span class="op">:</span><span class="dv">2017</span>)</span>
<span id="cb111-3"><a href="chapter12.html#cb111-3"></a>p2 &lt;-<span class="st"> </span><span class="kw">pmcplot</span>(terms, <span class="dv">2010</span><span class="op">:</span><span class="dv">2017</span>, <span class="dt">proportion=</span><span class="ot">FALSE</span>)</span>
<span id="cb111-4"><a href="chapter12.html#cb111-4"></a><span class="kw">plot_grid</span>(p, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<div class="figure"><span id="fig:pmcplot"></span>
<img src="clusterProfiler_files/figure-html/pmcplot-1.png" alt="Pmcplot of enrichment analysis." width="1344" />
<p class="caption">
Figure 12.18: <strong>Pmcplot of enrichment analysis.</strong>
</p>
</div>
</div>
<div id="goplot" class="section level2">
<h2><span class="header-section-number">12.10</span> goplot</h2>
<p><code>goplot</code> can accept output of <code>enrichGO</code> and visualized the enriched GO induced graph.</p>


<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="chapter12.html#cb112-1"></a><span class="kw">goplot</span>(ego)</span></code></pre></div>
</div>
<div id="browsekegg" class="section level2">
<h2><span class="header-section-number">12.11</span> browseKEGG</h2>
<p>To view the KEGG pathway, user can use <code>browseKEGG</code> function, which will open web browser and highlight enriched genes.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="chapter12.html#cb113-1"></a><span class="kw">browseKEGG</span>(kk, <span class="st">&#39;hsa04110&#39;</span>)</span></code></pre></div>
<p><img src="figures/browseKEGG.png" /></p>
</div>
<div id="pathview-from-pathview-package" class="section level2">
<h2><span class="header-section-number">12.12</span> pathview from pathview package</h2>
<p><a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> users can also use <code>pathview</code> from the <a href="https://www.bioconductor.org/packages/pathview">pathview</a><span class="citation">(Luo and Brouwer <a href="#ref-luo_pathview" role="doc-biblioref">2013</a>)</span> to visualize KEGG pathway.</p>
<p>The following example illustrate how to visualize “hsa04110” pathway, which was enriched in our previous analysis.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="chapter12.html#cb114-1"></a><span class="kw">library</span>(<span class="st">&quot;pathview&quot;</span>)</span>
<span id="cb114-2"><a href="chapter12.html#cb114-2"></a>hsa04110 &lt;-<span class="st"> </span><span class="kw">pathview</span>(<span class="dt">gene.data  =</span> geneList,</span>
<span id="cb114-3"><a href="chapter12.html#cb114-3"></a>                     <span class="dt">pathway.id =</span> <span class="st">&quot;hsa04110&quot;</span>,</span>
<span id="cb114-4"><a href="chapter12.html#cb114-4"></a>                     <span class="dt">species    =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb114-5"><a href="chapter12.html#cb114-5"></a>                     <span class="dt">limit      =</span> <span class="kw">list</span>(<span class="dt">gene=</span><span class="kw">max</span>(<span class="kw">abs</span>(geneList)), <span class="dt">cpd=</span><span class="dv">1</span>))</span></code></pre></div>
<p><img src="figures/hsa04110_pathview.png" /></p>
<p>For further information, please refer to the vignette of <a href="https://www.bioconductor.org/packages/pathview">pathview</a><span class="citation">(Luo and Brouwer <a href="#ref-luo_pathview" role="doc-biblioref">2013</a>)</span>.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-luo_pathview">
<p>Luo, Weijun, and Cory Brouwer. 2013. “Pathview: An R/Bioconductor Package for Pathway-Based Data Integration and Visualization.” <em>Bioinformatics</em> 29 (14): 1830–1. <a href="https://doi.org/10.1093/bioinformatics/btt285">https://doi.org/10.1093/bioinformatics/btt285</a>.</p>
</div>
<div id="ref-yu_reactomepa_2016">
<p>Yu, Guangchuang, and Qing-Yu He. 2016. “ReactomePA: An R/Bioconductor Package for Reactome Pathway Analysis and Visualization.” <em>Molecular BioSystems</em> 12 (2): 477–79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.</p>
</div>
<div id="ref-yu2012">
<p>Yu, Guangchuang, Le-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. “ClusterProfiler: An R Package for Comparing Biological Themes Among Gene Clusters.” <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284–87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.</p>
</div>
<div id="ref-yu_dose_2015">
<p>Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. “DOSE: An R/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.” <em>Bioinformatics</em> 31 (4): 608–9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="3">
<li id="fn3"><p><a href="https://www.biostars.org/p/375555" class="uri">https://www.biostars.org/p/375555</a><a href="chapter12.html#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter11.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter13.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
