<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta http-equiv="refresh" content="0;URL=https://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter2.html"/>
<link rel="next" href="chapter4.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.9/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#terminology"><i class="fa fa-check"></i><b>1.1</b> Terminology</a><ul>
<li class="chapter" data-level="1.1.1" data-path="chapter1.html"><a href="chapter1.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>1.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="1.1.2" data-path="chapter1.html"><a href="chapter1.html#gene-ontology-go"><i class="fa fa-check"></i><b>1.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="1.1.3" data-path="chapter1.html"><a href="chapter1.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>1.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="1.1.4" data-path="chapter1.html"><a href="chapter1.html#other-gene-sets"><i class="fa fa-check"></i><b>1.1.4</b> Other gene sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Functional Enrichment Analysis Methods</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#over-representation-analysis"><i class="fa fa-check"></i><b>2.1</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.3</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#input-data"><i class="fa fa-check"></i><b>3.1</b> Input data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#cell-marker"><i class="fa fa-check"></i><b>3.3</b> Cell Marker</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#msigdb-analysis"><i class="fa fa-check"></i><b>3.4</b> MSigDb analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="chapter4.html"><a href="chapter4.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Gene Ontology Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter5.html"><a href="chapter5.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="chapter5.html"><a href="chapter5.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chapter5.html"><a href="chapter5.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>5.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="chapter12.html"><a href="chapter12.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="chapter12.html"><a href="chapter12.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="chapter12.html"><a href="chapter12.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="chapter12.html"><a href="chapter12.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="chapter12.html"><a href="chapter12.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> dplyr verbs for clusterProfiler</a><ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#filter"><i class="fa fa-check"></i><b>13.1</b> filter</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#arrange"><i class="fa fa-check"></i><b>13.2</b> arrange</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#select"><i class="fa fa-check"></i><b>13.3</b> select</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#mutate"><i class="fa fa-check"></i><b>13.4</b> mutate</a></li>
<li class="chapter" data-level="13.5" data-path="chapter13.html"><a href="chapter13.html#slice"><i class="fa fa-check"></i><b>13.5</b> slice</a></li>
<li class="chapter" data-level="13.6" data-path="chapter13.html"><a href="chapter13.html#summarise"><i class="fa fa-check"></i><b>13.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="chapter14.html"><a href="chapter14.html"><i class="fa fa-check"></i><b>14</b> Useful utilities</a><ul>
<li class="chapter" data-level="14.1" data-path="chapter14.html"><a href="chapter14.html#bitr"><i class="fa fa-check"></i><b>14.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="14.1.1" data-path="chapter14.html"><a href="chapter14.html#bitr_kegg"><i class="fa fa-check"></i><b>14.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="chapter14.html"><a href="chapter14.html#setReadable"><i class="fa fa-check"></i><b>14.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter3" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Universal enrichment analysis</h1>
<p><a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but it’s still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (e.g. GO via BlastGO or KEGG via KAAS), unsupported ontologies/pathways or customized annotations.</p>
<p><a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> provides <code>enricher</code> function for hypergeometric test and <code>GSEA</code> function for gene set enrichment analysis that are designed to accept user defined annotation. They accept two additional parameters <em>TERM2GENE</em> and <em>TERM2NAME</em>. As indicated in the parameter names, <em>TERM2GENE</em> is a data.frame with first column of term ID and second column of corresponding mapped gene and <em>TERM2NAME</em> is a data.frame with first column of term ID and second column of corresponding term name. <em>TERM2NAME</em> is optional.</p>
<div id="input-data" class="section level2">
<h2><span class="header-section-number">3.1</span> Input data</h2>
<p>For over representation analysis, all we need is a gene vector, that is a vector of gene IDs. These gene IDs can be obtained by differential expression analysis (<em>e.g.</em> with <a href="http://www.bioconductor.org/packages/DESeq2">DESeq2</a> package).</p>
<p>For gene set enrichment analysis, we need a ranked list of genes. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides an example dataset <code>geneList</code> which was derived from <code>R</code> package <a href="http://bioconductor.org/packages/breastCancerMAINZ">breastCancerMAINZ</a> that contained 200 samples, including 29 samples in grade I, 136 samples in grade II and 35 samples in grade III. We computed the ratios of geometric means of grade III samples versus geometric means of grade I samples. Logarithm of these ratios (base 2) were stored in <code>geneList</code> dataset.</p>
<p>The <code>geneList</code> contains three features:</p>
<ol style="list-style-type: decimal">
<li>numeric vector: fold change or other type of numerical variable</li>
<li>named vector: every number was named by the corresponding gene ID</li>
<li>sorted vector: number should be sorted in decreasing order</li>
</ol>
<p>Suppose you are importing your own data from a <em>csv</em> file and the file contains two columns, one for gene ID (no duplicated allowed) and another one for fold change, you can prepare your own <code>geneList</code> via the following command:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="chapter3.html#cb5-1"></a>d &lt;-<span class="st"> </span><span class="kw">read.csv</span>(your_csv_file)</span>
<span id="cb5-2"><a href="chapter3.html#cb5-2"></a><span class="co">## assume that 1st column is ID</span></span>
<span id="cb5-3"><a href="chapter3.html#cb5-3"></a><span class="co">## 2nd column is fold change</span></span>
<span id="cb5-4"><a href="chapter3.html#cb5-4"></a></span>
<span id="cb5-5"><a href="chapter3.html#cb5-5"></a><span class="co">## feature 1: numeric vector</span></span>
<span id="cb5-6"><a href="chapter3.html#cb5-6"></a>geneList &lt;-<span class="st"> </span>d[,<span class="dv">2</span>]</span>
<span id="cb5-7"><a href="chapter3.html#cb5-7"></a></span>
<span id="cb5-8"><a href="chapter3.html#cb5-8"></a><span class="co">## feature 2: named vector</span></span>
<span id="cb5-9"><a href="chapter3.html#cb5-9"></a><span class="kw">names</span>(geneList) &lt;-<span class="st"> </span><span class="kw">as.character</span>(d[,<span class="dv">1</span>])</span>
<span id="cb5-10"><a href="chapter3.html#cb5-10"></a></span>
<span id="cb5-11"><a href="chapter3.html#cb5-11"></a><span class="co">## feature 3: decreasing order</span></span>
<span id="cb5-12"><a href="chapter3.html#cb5-12"></a>geneList &lt;-<span class="st"> </span><span class="kw">sort</span>(geneList, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>We can load the sample data into R via:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="chapter3.html#cb6-1"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb6-2"><a href="chapter3.html#cb6-2"></a><span class="kw">head</span>(geneList)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<p>Suppose we define fold change greater than 2 as DEGs:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="chapter3.html#cb8-1"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb8-2"><a href="chapter3.html#cb8-2"></a><span class="kw">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
</div>
<div id="wikipathways-analysis" class="section level2">
<h2><span class="header-section-number">3.2</span> WikiPathways analysis</h2>
<p><a href="https://www.wikipathways.org">WikiPathways</a> is a continuously updated pathway database curated by a community of researchers and pathway enthusiasts. WikiPathways produces monthly releases of gmt files for supported organisms at <a href="http://data.wikipathways.org/current/gmt/">data.wikipathways.org</a>. Download the appropriate gmt file and then generate <code>TERM2GENE</code> and <code>TERM2NAME</code> to use <code>enricher</code> and <code>GSEA</code> functions.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="chapter3.html#cb10-1"></a><span class="kw">library</span>(magrittr)</span>
<span id="cb10-2"><a href="chapter3.html#cb10-2"></a><span class="kw">library</span>(clusterProfiler)</span>
<span id="cb10-3"><a href="chapter3.html#cb10-3"></a></span>
<span id="cb10-4"><a href="chapter3.html#cb10-4"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb10-5"><a href="chapter3.html#cb10-5"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb10-6"><a href="chapter3.html#cb10-6"></a></span>
<span id="cb10-7"><a href="chapter3.html#cb10-7"></a>wpgmtfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/wikipathways-20180810-gmt-Homo_sapiens.gmt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;clusterProfiler&quot;</span>)</span>
<span id="cb10-8"><a href="chapter3.html#cb10-8"></a>wp2gene &lt;-<span class="st"> </span><span class="kw">read.gmt</span>(wpgmtfile)</span>
<span id="cb10-9"><a href="chapter3.html#cb10-9"></a>wp2gene &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate</span>(ont, <span class="kw">c</span>(<span class="st">&quot;name&quot;</span>,<span class="st">&quot;version&quot;</span>,<span class="st">&quot;wpid&quot;</span>,<span class="st">&quot;org&quot;</span>), <span class="st">&quot;%&quot;</span>)</span>
<span id="cb10-10"><a href="chapter3.html#cb10-10"></a>wpid2gene &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(wpid, gene) <span class="co">#TERM2GENE</span></span>
<span id="cb10-11"><a href="chapter3.html#cb10-11"></a>wpid2name &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(wpid, name) <span class="co">#TERM2NAME</span></span>
<span id="cb10-12"><a href="chapter3.html#cb10-12"></a></span>
<span id="cb10-13"><a href="chapter3.html#cb10-13"></a>ewp &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE =</span> wpid2gene, <span class="dt">TERM2NAME =</span> wpid2name)</span>
<span id="cb10-14"><a href="chapter3.html#cb10-14"></a><span class="kw">head</span>(ewp)</span></code></pre></div>
<pre><code>##            ID
## WP2446 WP2446
## WP2361 WP2361
## WP179   WP179
## WP3942 WP3942
## WP4240 WP4240
## WP2328 WP2328
##                                                                           Description
## WP2446                                                  Retinoblastoma (RB) in Cancer
## WP2361                                                       Gastric Cancer Network 1
## WP179                                                                      Cell Cycle
## WP3942                                                         PPAR signaling pathway
## WP4240 Regulation of sister chromatid separation at the metaphase-anaphase transition
## WP2328                                                            Allograft Rejection
##        GeneRatio  BgRatio       pvalue     p.adjust
## WP2446     11/95  88/6249 6.801697e-08 1.054263e-05
## WP2361      6/95  29/6249 3.772735e-06 2.923870e-04
## WP179      10/95 122/6249 1.384549e-05 7.153503e-04
## WP3942      7/95  67/6249 6.210513e-05 2.406574e-03
## WP4240      4/95  16/6249 7.931988e-05 2.458916e-03
## WP2328      7/95  90/6249 4.016758e-04 1.037663e-02
##              qvalue
## WP2446 9.450779e-06
## WP2361 2.621058e-04
## WP179  6.412648e-04
## WP3942 2.157336e-03
## WP4240 2.204258e-03
## WP2328 9.301967e-03
##                                                       geneID
## WP2446 8318/9133/7153/6241/890/983/81620/7272/1111/891/24137
## WP2361                       4605/7153/11065/22974/6286/6790
## WP179          8318/991/9133/890/983/7272/1111/891/4174/9232
## WP3942                    4312/9415/9370/5105/2167/3158/5346
## WP4240                                    991/1062/4085/9232
## WP2328                   10563/6373/4283/3002/10578/3117/730
##        Count
## WP2446    11
## WP2361     6
## WP179     10
## WP3942     7
## WP4240     4
## WP2328     7</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="chapter3.html#cb12-1"></a>ewp2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> wpid2gene, <span class="dt">TERM2NAME =</span> wpid2name, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb12-2"><a href="chapter3.html#cb12-2"></a><span class="kw">head</span>(ewp2)</span></code></pre></div>
<pre><code>##            ID
## WP4172 WP4172
## WP3932 WP3932
## WP306   WP306
## WP236   WP236
## WP474   WP474
## WP2911 WP2911
##                                           Description
## WP4172                     PI3K-Akt Signaling Pathway
## WP3932 Focal Adhesion-PI3K-Akt-mTOR-signaling pathway
## WP306                                  Focal Adhesion
## WP236                                    Adipogenesis
## WP474                       Endochondral Ossification
## WP2911    miRNA targets in ECM and membrane receptors
##        setSize enrichmentScore       NES      pvalue
## WP4172     311      -0.3406070 -1.467847 0.001347709
## WP3932     281      -0.3903410 -1.664489 0.001383126
## WP306      188      -0.4230017 -1.731019 0.001449275
## WP236      125      -0.4387536 -1.694197 0.001526718
## WP474       59      -0.5111776 -1.751890 0.001572327
## WP2911      21      -0.6953475 -1.888706 0.001757469
##          p.adjust    qvalues rank
## WP4172 0.03001658 0.02487562 2133
## WP3932 0.03001658 0.02487562 1994
## WP306  0.03001658 0.02487562 2221
## WP236  0.03001658 0.02487562 2301
## WP474  0.03001658 0.02487562 2142
## WP2911 0.03001658 0.02487562 2629
##                          leading_edge
## WP4172 tags=24%, list=17%, signal=21%
## WP3932 tags=26%, list=16%, signal=22%
## WP306  tags=28%, list=18%, signal=23%
## WP236  tags=34%, list=18%, signal=28%
## WP474  tags=47%, list=17%, signal=40%
## WP2911 tags=76%, list=21%, signal=60%
##                                                                                                                                                                                                                                                                                                                                                                                        core_enrichment
## WP4172 5228/7424/7157/627/2252/7059/92579/5563/5295/6794/1288/7010/3910/3371/3082/1291/4602/3791/1027/90993/3441/3643/1129/2322/1975/7450/596/3685/1942/2149/1280/4804/3675/595/2261/7248/2246/4803/2259/3912/1902/1278/1277/2846/2057/1293/2247/55970/5618/7058/10161/56034/3693/4254/3480/4908/5159/1292/3908/2690/3909/8817/4915/3551/2791/63923/3913/3667/1287/3679/7060/3479/80310/1311/5105/1101
## WP3932           2252/7059/51719/8660/5563/5295/6794/1288/7010/3910/3371/3082/3791/1301/1027/90993/3643/1129/1975/7450/3685/2034/1942/2149/1280/4804/3675/2261/7248/2246/4803/2259/3912/1902/2308/1278/1277/81617/2846/2057/2247/1281/50509/1290/55970/5618/7058/10161/56034/3693/4254/6720/3480/5159/3991/1289/1292/3908/2690/3909/8817/3551/2791/63923/3913/3667/3679/7060/3479/80310/1311/1101/3169
## WP306                                                                                                                       10420/5595/5228/7424/1499/4636/83660/7059/5295/1288/23396/3910/3371/3082/394/3791/7450/596/3685/1280/3675/595/2318/3912/1793/1278/5753/1277/10398/55742/50509/1290/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/3679/7060/3479/10451/80310/1311/1101
## WP236                                                                                                                                                                                 9611/8321/5925/8609/1499/2908/4088/8660/3399/6778/8648/6258/2662/5914/6776/2034/196/8204/4692/4208/2308/5468/6695/4023/1592/7350/81029/3952/1675/5618/6720/3991/2487/3667/3572/3479/6424/9370/5105/652/5346/2625
## WP474                                                                                                                                                                                                                                                    9759/4921/51719/860/5745/1028/1280/2261/4256/7042/4208/4322/8100/7078/2247/11096/85477/3480/2690/8817/5167/2737/5744/2487/5327/1300/1473/3479
## WP2911                                                                                                                                                                                                                                                                                                                6383/3672/7057/3915/3910/1291/1278/1293/1281/50509/1290/7058/3693/1289/1292/3913</code></pre>
<p>You may want to convert the gene IDs to gene symbols, which can be done by <code>setReadable</code> function.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="chapter3.html#cb14-1"></a><span class="kw">library</span>(org.Hs.eg.db)</span>
<span id="cb14-2"><a href="chapter3.html#cb14-2"></a>ewp &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ewp, org.Hs.eg.db, <span class="dt">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb14-3"><a href="chapter3.html#cb14-3"></a>ewp2 &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ewp2, org.Hs.eg.db, <span class="dt">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb14-4"><a href="chapter3.html#cb14-4"></a><span class="kw">head</span>(ewp)</span></code></pre></div>
<pre><code>##            ID
## WP2446 WP2446
## WP2361 WP2361
## WP179   WP179
## WP3942 WP3942
## WP4240 WP4240
## WP2328 WP2328
##                                                                           Description
## WP2446                                                  Retinoblastoma (RB) in Cancer
## WP2361                                                       Gastric Cancer Network 1
## WP179                                                                      Cell Cycle
## WP3942                                                         PPAR signaling pathway
## WP4240 Regulation of sister chromatid separation at the metaphase-anaphase transition
## WP2328                                                            Allograft Rejection
##        GeneRatio  BgRatio       pvalue     p.adjust
## WP2446     11/95  88/6249 6.801697e-08 1.054263e-05
## WP2361      6/95  29/6249 3.772735e-06 2.923870e-04
## WP179      10/95 122/6249 1.384549e-05 7.153503e-04
## WP3942      7/95  67/6249 6.210513e-05 2.406574e-03
## WP4240      4/95  16/6249 7.931988e-05 2.458916e-03
## WP2328      7/95  90/6249 4.016758e-04 1.037663e-02
##              qvalue
## WP2446 9.450779e-06
## WP2361 2.621058e-04
## WP179  6.412648e-04
## WP3942 2.157336e-03
## WP4240 2.204258e-03
## WP2328 9.301967e-03
##                                                              geneID
## WP2446 CDC45/CCNB2/TOP2A/RRM2/CCNA2/CDK1/CDT1/TTK/CHEK1/CCNB1/KIF4A
## WP2361                           MYBL2/TOP2A/UBE2C/TPX2/S100P/AURKA
## WP179       CDC45/CDC20/CCNB2/CCNA2/CDK1/TTK/CHEK1/CCNB1/MCM5/PTTG1
## WP3942                    MMP1/FADS2/ADIPOQ/PCK1/FABP4/HMGCS2/PLIN1
## WP4240                                     CDC20/CENPE/MAD2L1/PTTG1
## WP2328                    CXCL13/CXCL11/CXCL9/GZMB/GNLY/HLA-DQA1/C7
##        Count
## WP2446    11
## WP2361     6
## WP179     10
## WP3942     7
## WP4240     4
## WP2328     7</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="chapter3.html#cb16-1"></a><span class="kw">head</span>(ewp2)</span></code></pre></div>
<pre><code>##            ID
## WP4172 WP4172
## WP3932 WP3932
## WP306   WP306
## WP236   WP236
## WP474   WP474
## WP2911 WP2911
##                                           Description
## WP4172                     PI3K-Akt Signaling Pathway
## WP3932 Focal Adhesion-PI3K-Akt-mTOR-signaling pathway
## WP306                                  Focal Adhesion
## WP236                                    Adipogenesis
## WP474                       Endochondral Ossification
## WP2911    miRNA targets in ECM and membrane receptors
##        setSize enrichmentScore       NES      pvalue
## WP4172     311      -0.3406070 -1.467847 0.001347709
## WP3932     281      -0.3903410 -1.664489 0.001383126
## WP306      188      -0.4230017 -1.731019 0.001449275
## WP236      125      -0.4387536 -1.694197 0.001526718
## WP474       59      -0.5111776 -1.751890 0.001572327
## WP2911      21      -0.6953475 -1.888706 0.001757469
##          p.adjust    qvalues rank
## WP4172 0.03001658 0.02487562 2133
## WP3932 0.03001658 0.02487562 1994
## WP306  0.03001658 0.02487562 2221
## WP236  0.03001658 0.02487562 2301
## WP474  0.03001658 0.02487562 2142
## WP2911 0.03001658 0.02487562 2629
##                          leading_edge
## WP4172 tags=24%, list=17%, signal=21%
## WP3932 tags=26%, list=16%, signal=22%
## WP306  tags=28%, list=18%, signal=23%
## WP236  tags=34%, list=18%, signal=28%
## WP474  tags=47%, list=17%, signal=40%
## WP2911 tags=76%, list=21%, signal=60%
##                                                                                                                                                                                                                                                                                                                                                                                                                                      core_enrichment
## WP4172 PGF/VEGFC/TP53/BDNF/FGF7/THBS3/G6PC3/PRKAA2/PIK3R1/STK11/COL4A6/TEK/LAMA4/TNC/HGF/COL6A1/MYB/KDR/CDKN1B/CREB3L1/IFNA4/INSR/CHRM2/FLT3/EIF4B/VWF/BCL2/ITGAV/EFNA1/F2R/COL2A1/NGFR/ITGA3/CCND1/FGFR3/TSC1/FGF1/NGF/FGF14/LAMB1/LPAR1/COL1A2/COL1A1/LPAR4/EPOR/COL6A3/FGF2/GNG12/PRLR/THBS2/LPAR6/PDGFC/ITGB5/KITLG/IGF1R/NTF3/PDGFRB/COL6A2/LAMA2/GHR/LAMA3/FGF18/NTRK2/IKBKB/GNG11/TNN/LAMB2/IRS1/COL4A5/ITGA7/THBS4/IGF1/PDGFD/COMP/PCK1/CHAD
## WP3932      FGF7/THBS3/CAB39/IRS2/PRKAA2/PIK3R1/STK11/COL4A6/TEK/LAMA4/TNC/HGF/KDR/COL11A1/CDKN1B/CREB3L1/INSR/CHRM2/EIF4B/VWF/ITGAV/EPAS1/EFNA1/F2R/COL2A1/NGFR/ITGA3/FGFR3/TSC1/FGF1/NGF/FGF14/LAMB1/LPAR1/FOXO1/COL1A2/COL1A1/CAB39L/LPAR4/EPOR/FGF2/COL3A1/COL5A3/COL5A2/GNG12/PRLR/THBS2/LPAR6/PDGFC/ITGB5/KITLG/SREBF1/IGF1R/PDGFRB/LIPE/COL5A1/COL6A2/LAMA2/GHR/LAMA3/FGF18/IKBKB/GNG11/TNN/LAMB2/IRS1/ITGA7/THBS4/IGF1/PDGFD/COMP/CHAD/FOXA1
## WP306                                                                                                                                   TESK2/MAPK3/PGF/VEGFC/CTNNB1/MYL5/TLN2/THBS3/PIK3R1/COL4A6/PIP5K1C/LAMA4/TNC/HGF/ARHGAP5/KDR/VWF/BCL2/ITGAV/COL2A1/ITGA3/CCND1/FLNC/LAMB1/DOCK1/COL1A2/PTK6/COL1A1/MYL9/PARVA/COL5A3/COL5A2/FLNB/THBS2/SHC2/PDGFC/ITGB5/IGF1R/PDGFRB/CAV1/COL6A2/LAMA2/LAMA3/TNN/LAMB2/ITGA7/THBS4/IGF1/VAV3/PDGFD/COMP/CHAD
## WP236                                                                                                                                                                                                        NCOR1/FZD1/RB1/KLF7/CTNNB1/NR3C1/SMAD3/IRS2/ID3/STAT6/NCOA1/RXRG/GDF10/RARA/STAT5A/EPAS1/AHR/NRIP1/NDN/MEF2C/FOXO1/PPARG/SPOCK1/LPL/CYP26A1/UCP1/WNT5B/LEP/CFD/PRLR/SREBF1/LIPE/FRZB/IRS1/IL6ST/IGF1/SFRP4/ADIPOQ/PCK1/BMP4/PLIN1/GATA3
## WP474                                                                                                                                                                                                                                                                              HDAC4/DDR2/CAB39/RUNX2/PTH1R/CDKN1C/COL2A1/FGFR3/MGP/TGFB2/MEF2C/MMP13/IFT88/TIMP3/FGF2/ADAMTS5/SCIN/IGF1R/GHR/FGF18/ENPP1/GLI3/PTHLH/FRZB/PLAT/COL10A1/CST5/IGF1
## WP2911                                                                                                                                                                                                                                                                                                                                        SDC2/ITGA1/THBS1/LAMC1/LAMA4/COL6A1/COL1A2/COL6A3/COL3A1/COL5A3/COL5A2/THBS2/ITGB5/COL5A1/COL6A2/LAMB2</code></pre>
<p>As an alternative to manually downloading gmt files, install the <a href="https://bioconductor.org/packages/release/bioc/html/rWikiPathways.html">rWikiPathways package</a> to gain scripting access to the latest gmt files using the <code>downloadPathwayArchive</code> function.</p>
</div>
<div id="cell-marker" class="section level2">
<h2><span class="header-section-number">3.3</span> Cell Marker</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="chapter3.html#cb18-1"></a>cell_markers &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(<span class="st">&#39;http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb18-2"><a href="chapter3.html#cb18-2"></a><span class="st">   </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;cellMarker&quot;</span>, tissueType, cancerType, cellName, <span class="dt">sep=</span><span class="st">&quot;, &quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-3"><a href="chapter3.html#cb18-3"></a><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">select</span>(cellMarker, geneID) <span class="op">%&gt;%</span></span>
<span id="cb18-4"><a href="chapter3.html#cb18-4"></a><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneID =</span> <span class="kw">strsplit</span>(geneID, <span class="st">&#39;, &#39;</span>))</span>
<span id="cb18-5"><a href="chapter3.html#cb18-5"></a>cell_markers</span></code></pre></div>
<pre><code>## # A tibble: 2,868 x 2
##    cellMarker                                       geneID  
##    &lt;chr&gt;                                            &lt;list&gt;  
##  1 Kidney, Normal, Proximal tubular cell            &lt;chr [1…
##  2 Liver, Normal, Ito cell (hepatic stellate cell)  &lt;chr [1…
##  3 Endometrium, Normal, Trophoblast cell            &lt;chr [1…
##  4 Germ, Normal, Primordial germ cell               &lt;chr [1…
##  5 Corneal epithelium, Normal, Epithelial cell      &lt;chr [1…
##  6 Placenta, Normal, Cytotrophoblast                &lt;chr [1…
##  7 Periosteum, Normal, Periosteum-derived progenit… &lt;chr [4…
##  8 Amniotic membrane, Normal, Amnion epithelial ce… &lt;chr [2…
##  9 Primitive streak, Normal, Primitive streak cell  &lt;chr [2…
## 10 Adipose tissue, Normal, Stromal vascular fracti… &lt;chr [1…
## # … with 2,858 more rows</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="chapter3.html#cb20-1"></a>y &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>cell_markers, <span class="dt">minGSSize=</span><span class="dv">1</span>)</span>
<span id="cb20-2"><a href="chapter3.html#cb20-2"></a>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(y))</span></code></pre></div>
<div id="htmlwidget-fe3dd9c407c54050837d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fe3dd9c407c54050837d">{"x":{"filter":"none","data":[["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["33/153","20/153","6/153","7/153","6/153","9/153"],["166/11582","111/11582","42/11582","99/11582","77/11582","209/11582"],[4.43518066168319e-30,1.42606245514518e-17,1.70645935795602e-05,0.000329875455341899,0.0005276542867749,0.00181179940206466],[6.34230834620697e-28,1.0196346554288e-15,0.000813412293959037,0.0117930475284729,0.0150909126017621,0.0431812190825411],[4.80866955950915e-28,7.7307596252607e-16,0.000616720399542001,0.00894136102637251,0.0114417666395399,0.0327395330548526],["991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/1602/2018/4137","991/9133/10403/7153/259266/6241/9787/11065/51203/22974/10460/890/983/332/9212/9055/891/9232/2167/51705","6280/6279/597/7850/6286/366","1307/730/54829/10234/79901/57758/4969","6280/6279/597/7850/6286/366","9493/23397/9787/10460/55839/332/7272/4521/9232"],[33,20,6,7,6,9]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[5,6,7,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="msigdb-analysis" class="section level2">
<h2><span class="header-section-number">3.4</span> MSigDb analysis</h2>
<p><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a> contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can download GMT files from <a href="http://software.broadinstitute.org/gsea/msigdb">Broad Institute</a> and use <code>read.gmt</code> to parse the file to be used in <code>enricher()</code> and <code>GSEA()</code>.</p>
<p>There is an R package, <a href="https://cran.r-project.org/package=msigdbr">msigdbr</a>, that already packed the MSigDB gene sets in tidy data format that can be used directly with <em>clusterProfiler</em>.</p>
<p>It supports several specices:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="chapter3.html#cb21-1"></a><span class="kw">library</span>(msigdbr)</span>
<span id="cb21-2"><a href="chapter3.html#cb21-2"></a><span class="kw">msigdbr_show_species</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;Bos taurus&quot;               &quot;Caenorhabditis elegans&quot;  
##  [3] &quot;Canis lupus familiaris&quot;   &quot;Danio rerio&quot;             
##  [5] &quot;Drosophila melanogaster&quot;  &quot;Gallus gallus&quot;           
##  [7] &quot;Homo sapiens&quot;             &quot;Mus musculus&quot;            
##  [9] &quot;Rattus norvegicus&quot;        &quot;Saccharomyces cerevisiae&quot;
## [11] &quot;Sus scrofa&quot;</code></pre>
<p>We can retrieve all human gene sets:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="chapter3.html#cb23-1"></a>m_df &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span>
<span id="cb23-2"><a href="chapter3.html#cb23-2"></a><span class="kw">head</span>(m_df, <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</span></code></pre></div>
<pre><code>##          gs_name  gs_id gs_cat gs_subcat human_gene_symbol
## 1 AAACCAC_MIR140 M12609     C3       MIR             ABCC4
## 2 AAACCAC_MIR140 M12609     C3       MIR          ABRAXAS2
##   species_name entrez_gene gene_symbol sources
## 1 Homo sapiens       10257       ABCC4    &lt;NA&gt;
## 2 Homo sapiens       23172    ABRAXAS2    &lt;NA&gt;</code></pre>
<p>Or specific collection. Here we use C6, oncogenic gene sets as an example:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="chapter3.html#cb25-1"></a>m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C6&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb25-2"><a href="chapter3.html#cb25-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)</span>
<span id="cb25-3"><a href="chapter3.html#cb25-3"></a><span class="kw">head</span>(m_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name              entrez_gene
##   &lt;chr&gt;                      &lt;int&gt;
## 1 AKT_UP_MTOR_DN.V1_DN       25864
## 2 AKT_UP_MTOR_DN.V1_DN          95
## 3 AKT_UP_MTOR_DN.V1_DN      137872
## 4 AKT_UP_MTOR_DN.V1_DN         134
## 5 AKT_UP_MTOR_DN.V1_DN       55326
## 6 AKT_UP_MTOR_DN.V1_DN         216</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="chapter3.html#cb27-1"></a>em &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>m_t2g)</span>
<span id="cb27-2"><a href="chapter3.html#cb27-2"></a>em2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)</span>
<span id="cb27-3"><a href="chapter3.html#cb27-3"></a><span class="kw">head</span>(em)</span></code></pre></div>
<pre><code>##                                            ID
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN
## CSR_LATE_UP.V1_UP           CSR_LATE_UP.V1_UP
## ATF2_S_UP.V1_UP               ATF2_S_UP.V1_UP
##                                   Description GeneRatio
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN    22/185
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP    16/185
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN    15/185
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN    14/185
## CSR_LATE_UP.V1_UP           CSR_LATE_UP.V1_UP    12/185
## ATF2_S_UP.V1_UP               ATF2_S_UP.V1_UP    12/185
##                          BgRatio       pvalue     p.adjust
## RPS14_DN.V1_DN         185/10962 4.855041e-13 8.205020e-11
## GCNP_SHH_UP_LATE.V1_UP 186/10962 9.357887e-08 7.907414e-06
## VEGF_A_UP.V1_DN        193/10962 8.887046e-07 5.006369e-05
## PRC2_EZH2_UP.V1_DN     190/10962 3.907976e-06 1.651120e-04
## CSR_LATE_UP.V1_UP      166/10962 2.368079e-05 8.004105e-04
## ATF2_S_UP.V1_UP        190/10962 8.879028e-05 2.143651e-03
##                              qvalue
## RPS14_DN.V1_DN         6.694846e-11
## GCNP_SHH_UP_LATE.V1_UP 6.452017e-06
## VEGF_A_UP.V1_DN        4.084923e-05
## PRC2_EZH2_UP.V1_DN     1.347223e-04
## CSR_LATE_UP.V1_UP      6.530911e-04
## ATF2_S_UP.V1_UP        1.749102e-03
##                                                                                                                                        geneID
## RPS14_DN.V1_DN         10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/5105/2532/7021/79901
## GCNP_SHH_UP_LATE.V1_UP                                      55388/7153/79733/6241/9787/51203/983/9212/1111/9319/9055/3833/6790/4174/3169/1580
## VEGF_A_UP.V1_DN                                                   8318/9493/1062/9133/10403/6241/9787/4085/332/3832/7272/891/23362/2167/10234
## PRC2_EZH2_UP.V1_DN                                               8318/55388/4605/23397/9787/55355/10460/81620/2146/7272/9212/11182/3887/24137
## CSR_LATE_UP.V1_UP                                                               55143/2305/4605/6241/11065/55872/983/332/2146/51659/9319/1580
## ATF2_S_UP.V1_UP                                                               4312/6280/2305/9493/7153/2146/10112/3887/8544/22885/10234/10974
##                        Count
## RPS14_DN.V1_DN            22
## GCNP_SHH_UP_LATE.V1_UP    16
## VEGF_A_UP.V1_DN           15
## PRC2_EZH2_UP.V1_DN        14
## CSR_LATE_UP.V1_UP         12
## ATF2_S_UP.V1_UP           12</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="chapter3.html#cb29-1"></a><span class="kw">head</span>(em2)</span></code></pre></div>
<pre><code>##                              ID     Description setSize
## LEF1_UP.V1_DN     LEF1_UP.V1_DN   LEF1_UP.V1_DN     186
## LTE2_UP.V1_UP     LTE2_UP.V1_UP   LTE2_UP.V1_UP     183
## RAF_UP.V1_DN       RAF_UP.V1_DN    RAF_UP.V1_DN     189
## IL2_UP.V1_DN       IL2_UP.V1_DN    IL2_UP.V1_DN     182
## ATF2_S_UP.V1_DN ATF2_S_UP.V1_DN ATF2_S_UP.V1_DN     181
## ATF2_UP.V1_DN     ATF2_UP.V1_DN   ATF2_UP.V1_DN     177
##                 enrichmentScore       NES      pvalue
## LEF1_UP.V1_DN        -0.4857056 -1.987140 0.001381215
## LTE2_UP.V1_UP        -0.3938357 -1.608416 0.001383126
## RAF_UP.V1_DN         -0.5521243 -2.254389 0.001383126
## IL2_UP.V1_DN         -0.4252756 -1.732758 0.001388889
## ATF2_S_UP.V1_DN      -0.4460160 -1.815767 0.001394700
## ATF2_UP.V1_DN        -0.5008413 -2.032113 0.001402525
##                   p.adjust     qvalues rank
## LEF1_UP.V1_DN   0.01354839 0.007394831 1705
## LTE2_UP.V1_UP   0.01354839 0.007394831 2110
## RAF_UP.V1_DN    0.01354839 0.007394831 2035
## IL2_UP.V1_DN    0.01354839 0.007394831 2774
## ATF2_S_UP.V1_DN 0.01354839 0.007394831 2531
## ATF2_UP.V1_DN   0.01354839 0.007394831 2055
##                                   leading_edge
## LEF1_UP.V1_DN   tags=35%, list=14%, signal=31%
## LTE2_UP.V1_UP   tags=33%, list=17%, signal=28%
## RAF_UP.V1_DN    tags=43%, list=16%, signal=37%
## IL2_UP.V1_DN    tags=39%, list=22%, signal=31%
## ATF2_S_UP.V1_DN tags=39%, list=20%, signal=32%
## ATF2_UP.V1_DN   tags=38%, list=16%, signal=32%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                  core_enrichment
## LEF1_UP.V1_DN                                                                        79153/22998/10325/23221/79170/9187/4300/51479/25956/7552/8644/11162/7168/1490/54463/65084/25837/80303/1809/3397/4208/11223/79762/6451/481/54795/51626/8991/5950/5627/64699/6414/10103/221078/9961/5874/56898/90865/83989/5002/6653/56034/55314/85458/956/54502/4832/5783/1363/53832/54985/80221/214/50853/51149/57088/5157/2947/79932/7031/6505/9338/25924/80736/3169/10551
## LTE2_UP.V1_UP                                                                                                       23074/64759/1601/6558/7138/11138/10497/81031/665/2878/3554/4925/55005/9252/2034/5283/54869/5025/6019/65124/10979/51435/26999/4212/444/23588/55556/2954/93129/79762/54453/79789/316/79679/151011/51280/9630/57037/5627/9867/55198/6653/10455/5101/65055/8309/9429/23303/9590/253190/4982/4128/4680/1846/51760/80129/4857/5507/3158/5304/10974
## RAF_UP.V1_DN    23314/221037/1960/51340/55105/10265/22996/9687/5357/1952/12/4602/9231/596/51454/57419/595/4256/23022/8777/7837/7042/3397/57613/323/1831/6451/54843/7358/2353/8773/6938/8991/64699/57007/23389/10560/7227/3485/79068/10769/4254/26018/3480/2674/23327/857/116039/6542/2690/1955/1363/26353/7033/5376/89927/51363/8821/2239/6947/4886/214/3487/3667/5157/54847/54898/7031/6505/57535/10451/18/771/80129/5174/5507/56521/8839/8614/5241/10551/57758
## IL2_UP.V1_DN                                                  51276/10858/5333/2977/79874/10014/7773/64798/26959/6588/579/777/5158/23361/9915/5087/64759/2535/6123/7010/51285/126353/80201/79841/1028/2322/4222/947/9187/225/9976/5592/54996/3202/55779/23148/5334/8605/57326/26115/54453/55766/64927/6591/55204/9940/57608/653483/9886/10279/55805/10628/901/9891/7102/27244/79921/4674/8292/5125/79843/6857/79940/23541/3357/50853/9863/79932/90362/55663/4250
## ATF2_S_UP.V1_DN                                                               8714/53616/55667/51279/4642/7130/7139/2/710/80352/4608/11145/881/2252/3399/22846/10687/12/1397/1301/3554/2619/26011/78987/9284/1306/6586/9627/4054/54813/115207/55890/26249/4803/11075/3400/79789/8527/2353/10580/79987/80760/5493/90865/3485/3751/2202/2199/10610/1047/1294/23492/3908/5364/658/5350/2121/2331/4330/9737/4982/57088/8490/9358/6424/1307/3708/125/1311/56521/10351
## ATF2_UP.V1_DN                                                                             11145/9738/2252/8660/10687/5357/80201/12/22998/3554/2619/23556/8840/9284/57419/5737/6586/9627/8863/10085/115207/55890/23015/629/3212/3400/50650/8527/2353/79987/80760/9668/1012/10614/5493/8553/23126/90865/3485/3751/2202/57212/2199/10610/3075/6387/22834/1909/10129/57161/5350/2121/2331/4330/2952/4982/57088/9358/6424/6038/51313/1311/56521/10351/5304/4239/79901</code></pre>
<p>We can test with other collections, for example, using C3 to test whether the genes are up/down-regulated by sharing specific motif.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="chapter3.html#cb31-1"></a>m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C3&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb31-2"><a href="chapter3.html#cb31-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)</span>
<span id="cb31-3"><a href="chapter3.html#cb31-3"></a><span class="kw">head</span>(m_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name        entrez_gene
##   &lt;chr&gt;                &lt;int&gt;
## 1 AAACCAC_MIR140       10257
## 2 AAACCAC_MIR140       23172
## 3 AAACCAC_MIR140          81
## 4 AAACCAC_MIR140          90
## 5 AAACCAC_MIR140        8754
## 6 AAACCAC_MIR140       11096</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="chapter3.html#cb33-1"></a>em3 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)</span>
<span id="cb33-2"><a href="chapter3.html#cb33-2"></a><span class="kw">head</span>(em3)</span></code></pre></div>
<pre><code>##                                                            ID
## GGATTA_PITX2_Q2                               GGATTA_PITX2_Q2
## TTTGCAC_MIR19A_MIR19B                   TTTGCAC_MIR19A_MIR19B
## CAGTATT_MIR200B_MIR200C_MIR429 CAGTATT_MIR200B_MIR200C_MIR429
## AAAYRNCTG_UNKNOWN                           AAAYRNCTG_UNKNOWN
## TTGCACT_MIR130A_MIR301_MIR130B TTGCACT_MIR130A_MIR301_MIR130B
## ATGTACA_MIR493                                 ATGTACA_MIR493
##                                                   Description
## GGATTA_PITX2_Q2                               GGATTA_PITX2_Q2
## TTTGCAC_MIR19A_MIR19B                   TTTGCAC_MIR19A_MIR19B
## CAGTATT_MIR200B_MIR200C_MIR429 CAGTATT_MIR200B_MIR200C_MIR429
## AAAYRNCTG_UNKNOWN                           AAAYRNCTG_UNKNOWN
## TTGCACT_MIR130A_MIR301_MIR130B TTGCACT_MIR130A_MIR301_MIR130B
## ATGTACA_MIR493                                 ATGTACA_MIR493
##                                setSize enrichmentScore
## GGATTA_PITX2_Q2                    455      -0.3365909
## TTTGCAC_MIR19A_MIR19B              397      -0.3672122
## CAGTATT_MIR200B_MIR200C_MIR429     380      -0.3607152
## AAAYRNCTG_UNKNOWN                  290      -0.4355277
## TTGCACT_MIR130A_MIR301_MIR130B     313      -0.3890571
## ATGTACA_MIR493                     251      -0.3980657
##                                      NES      pvalue
## GGATTA_PITX2_Q2                -1.495575 0.001230012
## TTTGCAC_MIR19A_MIR19B          -1.612649 0.001254705
## CAGTATT_MIR200B_MIR200C_MIR429 -1.581860 0.001257862
## AAAYRNCTG_UNKNOWN              -1.863237 0.001319261
## TTGCACT_MIR130A_MIR301_MIR130B -1.675646 0.001322751
## ATGTACA_MIR493                 -1.676300 0.001347709
##                                  p.adjust    qvalues rank
## GGATTA_PITX2_Q2                0.04123119 0.02860412 2464
## TTTGCAC_MIR19A_MIR19B          0.04123119 0.02860412 4078
## CAGTATT_MIR200B_MIR200C_MIR429 0.04123119 0.02860412 3186
## AAAYRNCTG_UNKNOWN              0.04123119 0.02860412 2952
## TTGCACT_MIR130A_MIR301_MIR130B 0.04123119 0.02860412 2861
## ATGTACA_MIR493                 0.04123119 0.02860412 3911
##                                                  leading_edge
## GGATTA_PITX2_Q2                tags=26%, list=20%, signal=22%
## TTTGCAC_MIR19A_MIR19B          tags=44%, list=33%, signal=30%
## CAGTATT_MIR200B_MIR200C_MIR429 tags=34%, list=25%, signal=26%
## AAAYRNCTG_UNKNOWN              tags=38%, list=24%, signal=29%
## TTGCACT_MIR130A_MIR301_MIR130B tags=36%, list=23%, signal=28%
## ATGTACA_MIR493                 tags=45%, list=31%, signal=32%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       core_enrichment
## GGATTA_PITX2_Q2                                                                                                                                                                                                                                                                                                                               25836/311/3199/4604/57556/8019/3321/57835/8929/6671/1760/10401/4430/9497/5530/6532/54894/23528/5228/5087/2295/6453/56171/3232/4921/3005/221037/56912/10265/10521/10497/23767/8322/37/23429/6561/4303/57082/845/94/51447/9922/6258/23243/2571/10253/4602/29119/55068/126393/7402/56675/6302/5624/7716/7779/596/4300/57804/79923/1280/100506658/9627/3675/26037/2261/2804/65084/11030/3216/57326/140890/23255/5396/9854/1907/54843/57037/862/23452/55152/1848/8835/89795/57007/64221/5136/7227/10324/55273/6310/2202/56034/956/54502/51474/4254/6925/6097/744/443/23446/4306/4915/4487/25803/4081/4223/10472/388677/55107/347902/3708/10742/51313/4857/730/2532/4036/3169
## TTTGCAC_MIR19A_MIR19B          22911/1605/659/10318/4299/6249/81609/57194/9686/10425/9839/23013/22936/989/51742/4154/55625/51460/51592/11127/3976/54980/9751/56937/6045/4204/27252/4862/27236/2115/5869/152006/57698/57018/23032/56145/10150/55054/4090/25852/10618/2060/7107/10771/7071/9044/10904/9752/6431/8473/83452/392/10395/6777/51230/22889/23236/55652/158471/6595/905/56146/4026/79665/5932/115/8325/546/10154/9655/23499/54521/26959/23612/3321/56929/54778/26060/56848/9098/107/10420/23001/29922/9759/29116/23405/94134/2295/6453/83660/51719/8654/11138/2186/23341/8038/860/2004/9256/6548/23621/23503/29/394/220594/831/92689/55082/54014/1901/23111/7552/27303/1490/51454/57419/66008/2318/6252/7248/4929/23041/9900/55727/91694/22863/1389/54796/3400/9649/23253/54453/7048/22862/22905/23365/79899/5412/60481/54838/89795/5793/23389/7227/6310/3625/26018/23492/2099/23116/744/775/90355/388/23414/11069/6857/26960/55638/10472/2152/2697/3131/7786/55107/9863/23261/3479/3708/10742/2018/4036/2066
## CAGTATT_MIR200B_MIR200C_MIR429                                                                                                                                                                                                                                                    7071/31/23291/5179/57092/9969/9865/8473/9644/10395/6383/5569/1392/4008/9647/8613/6405/4884/8491/9706/2263/6567/53353/64766/5310/546/5128/55145/54469/200576/8434/7003/687/3915/1456/3321/10499/23189/26060/55578/1998/51421/22864/10484/10140/23001/4661/253782/9807/83660/95681/1960/2908/51719/2186/22846/108/9710/80205/2104/9522/7320/23243/23258/4602/129642/2669/51439/7716/54014/23047/5099/2823/3340/1942/7552/1843/100506658/8204/10979/79618/23015/8848/4692/5334/57515/26249/6563/23122/6196/1902/4077/2494/10370/3953/51339/79365/54838/89795/8076/5577/3751/6310/1009/29994/10769/1501/57509/3899/2273/23116/4908/27244/775/5783/65055/4035/23414/2737/5744/6857/55800/10186/3131/9828/6935/4982/219654/1602/3708/4857
## AAAYRNCTG_UNKNOWN                                                                                                                                                                                                                                                                                                                                                                                                        598/1392/4008/11080/79893/4724/8515/8613/81847/4297/3218/27314/7025/6595/8516/29966/11334/90/54329/89797/120/783/4750/8289/5071/3386/8929/2776/5530/23564/25959/4776/607/8038/5926/1288/2104/4303/845/51447/23243/22902/3082/54879/5745/4211/9024/23047/1942/79625/27303/4628/55911/51585/3202/6252/23338/64112/7482/444/55128/26249/64236/3397/64393/23635/23253/4013/1278/5252/323/1831/10370/1384/5549/26575/1272/89795/81603/55799/8470/7227/54875/761/477/29951/10468/6444/4254/7102/744/2045/4915/658/23414/11069/55800/10186/10472/2205/1287/7122/3479/1846/2018/2066/3169/8614/79901
## TTGCACT_MIR130A_MIR301_MIR130B                                                                                                                                                                                                                                                                                                                                                   9044/1808/196441/10904/27430/192670/8473/392/2909/83482/4762/4297/23236/27314/5932/4884/9706/90/546/9655/84196/5562/8626/757/26959/26060/56848/9098/23328/113791/107/10420/23001/29922/4661/23405/94134/2295/60468/55914/11138/2186/23341/6500/108/8038/51257/860/8648/80205/2004/9256/23243/23503/23469/4602/220594/8452/51111/55082/54014/1901/23111/7552/57419/389136/171586/3213/3202/6252/7248/23181/7482/55727/5797/91694/22863/10612/23253/8829/7048/753/23365/79365/79899/5468/5793/7227/6310/3625/26018/2099/23116/744/90355/23446/4306/23414/11069/64061/90627/26960/4223/55638/10472/2697/7786/23261/3708/10742/2018/2066
## ATGTACA_MIR493                                                                                                                                                                                                                                                                                                                                                                     23013/8729/5170/10949/2258/55205/8600/149076/9912/23369/1452/50618/2549/11163/152006/7403/8554/10150/23196/54880/4836/6670/55288/27245/9044/196441/51105/9969/25972/57178/6383/7703/6400/605/27005/55119/905/9295/7082/89797/5930/8567/22871/5532/200576/8289/22986/200734/6671/9098/107/7337/23001/4131/1120/8556/10424/825/55187/5295/8322/23390/1387/8648/23429/2104/23243/2571/2915/394/6299/7474/596/9252/3340/7552/10217/23284/9162/6844/22982/4208/9649/54665/2308/10370/79365/8406/1012/23345/23389/6625/65997/7227/3778/1009/56034/9369/1501/3899/26018/2045/65055/5376/89927/2239/26960/27123/367/80206/23024/1602/26289</code></pre>
<!--


# DAVID functional analysis

[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) provides enrichment and GSEA analysis with GO, KEGG, DO and Reactome pathway supported internally, some user may prefer GO and KEGG analysis with DAVID[@huang_david_2007] and still attracted by the visualization methods provided by [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@paranjpe_genome_wid_2013]. To bridge the gap between DAVID and clusterProfiler, we implemented `enrichDAVID`. This function query enrichment analysis result from DAVID webserver via [RDAVIDWebService](https://www.bioconductor.org/packages/RDAVIDWebService)[@fresno_rdavidwebservice_2013] and stored the result as an `enrichResult` instance, so that we can use all the visualization functions in [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) to visualize DAVID results. `enrichDAVID` is fully compatible with `compareCluster` function and comparing enrichment results from different gene clusters is now available with DAVID.


```r
david <- enrichDAVID(gene = gene,
                     idType = "ENTREZ_GENE_ID",
                     listType = "Gene",
                     annotation = "KEGG_PATHWAY",
                     david.user = "clusterProfiler@hku.hk")
```

DAVID Web Service has the following limitations:

+ A job with more than 3000 genes to generate gene or term cluster report will not be handled by DAVID due to resource limit.
+ No more than 200 jobs in a day from one user or computer.
+ DAVID Team reserves right to suspend any improper uses of the web service without notice.

For more details, please refer to [http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html](http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html).

As user has limited usage, please [register](http://david.abcc.ncifcrf.gov/webservice/register.htm) and use your own user account to run `enrichDAVID`.


-->

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
