<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>References | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="References | clusterProfiler: universal enrichment tool for functional and comparative study">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="References | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="useful-utilities.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="2" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html"><i class="fa fa-check"></i><b>2</b> Gene Set Enrichment Analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.1</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="universal-enrichment-analysis.html"><a href="universal-enrichment-analysis.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="universal-enrichment-analysis.html"><a href="universal-enrichment-analysis.html#the-genelist-dataset"><i class="fa fa-check"></i><b>3.1</b> The <code>geneList</code> dataset</a></li>
<li class="chapter" data-level="3.2" data-path="universal-enrichment-analysis.html"><a href="universal-enrichment-analysis.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="disease-analysis.html"><a href="disease-analysis.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="disease-analysis.html"><a href="disease-analysis.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="disease-analysis.html"><a href="disease-analysis.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="disease-analysis.html"><a href="disease-analysis.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="disease-analysis.html"><a href="disease-analysis.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="disease-analysis.html"><a href="disease-analysis.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="disease-analysis.html"><a href="disease-analysis.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="go-analysis.html"><a href="go-analysis.html"><i class="fa fa-check"></i><b>5</b> GO Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="go-analysis.html"><a href="go-analysis.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="go-analysis.html"><a href="go-analysis.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="go-analysis.html"><a href="go-analysis.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="go-analysis.html"><a href="go-analysis.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="go-analysis.html"><a href="go-analysis.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="go-analysis.html"><a href="go-analysis.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="go-analysis.html"><a href="go-analysis.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="go-analysis.html"><a href="go-analysis.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="kegg-analysis.html"><a href="kegg-analysis.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="kegg-analysis.html"><a href="kegg-analysis.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="kegg-analysis.html"><a href="kegg-analysis.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="kegg-analysis.html"><a href="kegg-analysis.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="kegg-analysis.html"><a href="kegg-analysis.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="msigdb-analysis.html"><a href="msigdb-analysis.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="reactome-pathway-analysis.html"><a href="reactome-pathway-analysis.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="mesh-enrichment-analysis.html"><a href="mesh-enrichment-analysis.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="functional-enrichment-analysis-of-genomic-coordinations.html"><a href="functional-enrichment-analysis-of-genomic-coordinations.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="biological-theme-comparison.html"><a href="biological-theme-comparison.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="biological-theme-comparison.html"><a href="biological-theme-comparison.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="biological-theme-comparison.html"><a href="biological-theme-comparison.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="visualization-of-functional-enrichment-result.html"><a href="visualization-of-functional-enrichment-result.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="useful-utilities.html"><a href="useful-utilities.html"><i class="fa fa-check"></i><b>13</b> Useful utilities</a><ul>
<li class="chapter" data-level="13.1" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr-biological-id-translator"><i class="fa fa-check"></i><b>13.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="13.1.1" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr_kegg-converting-biological-ids-using-kegg-api"><i class="fa fa-check"></i><b>13.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references">
<div>
<p>A., Omer, Giovanni M. D., Thanos P. M., and Francesca D. C. 2016. “NCG 5.0: Updates of a Manually Curated Repository of Cancer Genes and Associated Properties from Cancer Mutational Screenings.” <em>Nucleic Acids Research</em> 44 (D1): D992–D999. <a href="https://doi.org/10.1093/nar/gkv1123">https://doi.org/10.1093/nar/gkv1123</a>.</p>
</div>
<div>
<p>Boyle, Elizabeth I, Shuai Weng, Jeremy Gollub, Heng Jin, David Botstein, J Michael Cherry, and Gavin Sherlock. 2004. “GO::TermFinder–open Source Software for Accessing Gene Ontology Information and Finding Significantly Enriched Gene Ontology Terms Associated with a List of Genes.” <em>Bioinformatics (Oxford, England)</em> 20 (18): 3710–5. <a href="https://doi.org/10.1093/bioinformatics/bth456">https://doi.org/10.1093/bioinformatics/bth456</a>.</p>
</div>
<div>
<p>Du, Pan, Gang Feng, Jared Flatow, Jie Song, Michelle Holko, Warren A. Kibbe, and Simon M. Lin. 2009. “From Disease Ontology to Disease-Ontology Lite: Statistical Methods to Adapt a General-Purpose Ontology for the Test of Gene-Ontology Associations.” <em>Bioinformatics</em> 25 (12): i63–i68. <a href="https://doi.org/10.1093/bioinformatics/btp193">https://doi.org/10.1093/bioinformatics/btp193</a>.</p>
</div>
<div>
<p>Janet, P., Núria Q.R., Àlex B., Jordi D.P., Anna B.M., Martin B., Ferran S., and Laura I. F. 2015. “DisGeNET: A Discovery Platform for the Dynamical Exploration of Human Diseases and Their Genes.” <em>Database</em> 2015 (March): bav028. <a href="https://doi.org/10.1093/database/bav028">https://doi.org/10.1093/database/bav028</a>.</p>
</div>
<div>
<p>Luo, Weijun, and Cory Brouwer. 2013. “Pathview: An R/Bioconductor Package for Pathway-Based Data Integration and Visualization.” <em>Bioinformatics</em> 29 (14): 1830–1. <a href="https://doi.org/10.1093/bioinformatics/btt285">https://doi.org/10.1093/bioinformatics/btt285</a>.</p>
</div>
<div>
<p>S., Alexey. n.d. “An Algorithm for Fast Preranked Gene Set Enrichment Analysis Using Cumulative Statistic Calculation.” <em>Biorxiv</em>. <a href="https://doi.org/10.1101/060012">https://doi.org/10.1101/060012</a>.</p>
</div>
<div>
<p>Schmidt, Marcus, Daniel B?hm, Christian von T?rne, Eric Steiner, Alexander Puhl, Henryk Pilch, Hans-Anton Lehr, Jan G. Hengstler, Heinz K?lbl, and Mathias Gehrmann. 2008. “The Humoral Immune System Has a Key Prognostic Impact in Node-Negative Breast Cancer.” <em>Cancer Research</em> 68 (13): 5405–13. <a href="https://doi.org/10.1158/0008-5472.CAN-07-5206">https://doi.org/10.1158/0008-5472.CAN-07-5206</a>.</p>
</div>
<div>
<p>Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</p>
</div>
<div>
<p>Yu, Guangchuang, and Qing-Yu He. 2016. “ReactomePA: An R/Bioconductor Package for Reactome Pathway Analysis and Visualization.” <em>Molecular BioSystems</em> 12 (2): 477–79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.</p>
</div>
<div>
<p>Yu, Guangchuang, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, and Shengqi Wang. 2010. “GOSemSim: An R Package for Measuring Semantic Similarity Among Go Terms and Gene Products.” <em>Bioinformatics</em> 26 (7): 976–78. <a href="https://doi.org/10.1093/bioinformatics/btq064">https://doi.org/10.1093/bioinformatics/btq064</a>.</p>
</div>
<div>
<p>Yu, Guangchuang, Le-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. “ClusterProfiler: An R Package for Comparing Biological Themes Among Gene Clusters.” <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284–87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.</p>
</div>
<div>
<p>Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. “ChIPseeker: An R/Bioconductor Package for Chip Peak Annotation, Comparison and Visualization.” <em>Bioinformatics</em> 31 (14): 2382–3. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.</p>
</div>
<div>
<p>Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. “DOSE: An R/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.” <em>Bioinformatics</em> 31 (4): 608–9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="useful-utilities.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
